RevenueCatUI/CustomerCenter/Data/SemanticVersion.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  SemanticVersion.swift
+//
+//  Created by JayShortway on 08/08/2024.
+
+#if CUSTOMER_CENTER_ENABLED
+
+import Foundation
+
+struct SemanticVersion: Comparable {
+    let major: UInt
+    let minor: UInt
+    let patch: UInt
+
+    init(major: UInt, minor: UInt, patch: UInt) {
+        self.major = major
+        self.minor = minor
+        self.patch = patch
+    }
+
+    init(_ version: String) throws {
+        let pattern = #"^(\d+)(?:\.(\d+))?(?:\.(\d+))?$"#
+        guard let regex = try? NSRegularExpression(pattern: pattern),
```

Comment:
my first thought was that we can use the new [RegexBuilder](https://developer.apple.com/documentation/RegexBuilder), but that's iOS 16+ and we're 15+... So close

Reply:
Yea that seems to be much better!